### PR TITLE
added examples to nodes(), edges() and neighbors()

### PR DESCRIPTION
**Pull request by XiaobenYin**
Hi, I'm happy to contribute as an Outreachy applicant. I added some basic examples to nodes(), edges and neighbors() under function. And I found the return type of neighbors() should be an iterator object rather than a list, as we need to use list() function to turn it as a list.

### CODE REVIEW

File: networkx/classes/function.py [hunk 1]
```diff
@@ -54,7 +54,15 @@
 
 
 def nodes(G):
-    """Returns an iterator over the graph nodes."""
+    """Returns an iterator over the graph nodes.
```
Comment (rossbar) [inline networkx/classes/function.py:57]: I know this wasn't the goal of this PR, but looking over this I think this is actually incorrect - a `NodeView` object is returned, which is technically not an Iterator in the Python sense. I think this first summary line should be updated to use more precise wording; maybe something like ``Returns a view of the nodes in `G` ``.

File: networkx/classes/function.py [hunk 2]
```diff
@@ -54,7 +54,15 @@
...
+
+    Examples
+    --------
+    >>> G = nx.Graph()
+    >>> G.add_nodes_from([1, 2, 3, 4, 5])
```
Comment (rossbar) [inline networkx/classes/function.py:62]: This is just a nit, but it feels a little circular to explicitly create a graph with a node list then immediately reproduce the nodes! This is of course correct, but I think you'd get the same effect with using a well-known graph constructor, e.g.

```suggestion
    >>> G = nx.path_graph(5)
```

**Review by rossbar — COMMENTED** [review]
Generally this looks good, thanks @XiaobenYin .

Looking at these functions got me thinking about a higher-level question - are users encourage to use these functions? It seems they all have corresponding graph methods that in many cases have more features. Is there a strong use-case for these functions? If not, we might consider replacing the docstrings with an even shorter summary describing the methods that they wrap. For example, the docstring of `nodes` could look something like `Returns G.nodes()`.

In each case these are one-line wrapper functions, so I think it might actually be better to just explicitly state what they do in the docstring rather than add Examples, Parameters, Returns, etc. Thoughts?

**Comment by MridulS** [comment]
Yeah I think instead of repeating the information we should just add a link to the base Graph class methods. Thanks for working on this @XiaobenYin !

**Pull request closed without merging**